src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,295 @@
+        }
+
+        System.out.println();
+        printDivider();
+    }
+
+    public static void readFile() throws FileNotFoundException {
+        File file = new File("duke.txt");
+        try {
+            boolean isExisting = file.createNewFile();
+        } catch (IOException e) {
+            printExceptionMessage(EXCEPTION_4);
+        }
+        Scanner scanner = new Scanner(file);
+
+        while (scanner.hasNext()) {
+            String line = scanner.nextLine();
+            String[] words = line.split(" ");
+            try {
+                Task currentTask = createTask(line.substring(2));
+                addTask(currentTask, true);
+                if (words[0].equals("1")) {
+                    markAsDone(String.valueOf(numberOfTasks), true);
+                }
+            } catch (DukeException e) {
+                printExceptionMessage(EXCEPTION_2);
+            }
+        }
+    }
+
+    public static void appendToFile(String command) throws IOException {
+        FileWriter fw = new FileWriter("duke.txt", true);
+        fw.write("0 " + command + "\n");
+        fw.close();
+    }
+
+    public static void rewriteFile() throws IOException {
+        FileWriter fw = new FileWriter("duke.txt");
+        String number;
+        for (Task task : tasks) {
+            if (task.isDone) {
+                number = "1";
+            } else {
+                number = "0";
+            }
+            if (task instanceof Todo) {
+                fw.write(number + " todo " + task.description + "\n");
+            } else if (task instanceof Deadline) {
+                fw.write(number + " deadline " + task.description + " /by " + ((Deadline) task).by + "\n");
+            } else if (task instanceof Event) {
+                fw.write(number + " event " + task.description + " /at " + ((Event) task).at + "\n");
+            }
+        }
+        fw.close();
+    }
+
+    public static void performAction(String command) throws DukeException{
+        String[] words = command.split(" ");
+        String firstWord = words[0];
+
+        if (firstWord.equals("mark")) {
+            //Mark task as done
+            markAsDone(words[1], false);
+        } else if (firstWord.equals("unmark")) {
+            //Mark task as undone
+            markAsUndone(words[1], false);
+        } else if (firstWord.equals("list")) {
+            //Print the list
+            printList();
+        } else if (firstWord.equals("todo") | firstWord.equals("deadline") | firstWord.equals("event")) {
+            //Add a task to the list
+            try {
+                Task currentTask = createTask(command);
+                addTask(currentTask, false);
+                appendToFile(command);
+            } catch (DukeException e) {
+                printExceptionMessage(EXCEPTION_1);
+            } catch (IOException e) {
+                printExceptionMessage(EXCEPTION_4);
+            }
+        } else if (firstWord.equals("delete")) {
```

Comment:
Avoid long methods, if a method is longer than 30 lines of code, try to split it up into methods to improve readability.

Code Quality Guideline: Avoid Long Methods